detex.l

```diff
@@ -665,6 +691,65 @@ yyless(int n)
 }
 #endif
 
+/******
+** LineBreak -- choses the proper way to break a line. If '-1' option is
+**      enabled we also want to output some source location information.
+******/
+
+void
+LineBreak()
+{
+	if (fWord) return;
+	if (fSrcLoc && fIsColumn0) {
+		printf("%s:%4d: ", fFileNames[csb], fFileLines[csb]);
+		fIsColumn0 = 0;
+	}
+	putchar('\n');
+	fFileLines[csb]++; fIsColumn0=1;
+}
+
+/******
+** Echo -- If we are at column 0 and have specified '-1'; output 
+**      the source location information.
+******/
+
+void
+Echo()
+{
+	if (fSrcLoc && fIsColumn0) {
+		printf("%s:%4d: ", fFileNames[csb], fFileLines[csb]);
```

Comment:
Why do you left-pad line number with spaces? You still get uneven indentation (see with-srcloc-correct.txt: text begins in column 23 in line 6, but in column 12 in line 8) but apart from it you disconnect file name from line number. If you have both connected (like `input.tex:147`) then passing this combination as file name to some editors causes the file to be opened with the given line highlighted.

For me: you should either left-pad the _combination_ of file name + colon + line number or not left-pad anything (and simply use `%s%d`).

Reply:
Good question; I added it there to align the output a slightly better but it makes no sense. :)